Fix misuse of xmlnode_get_data()

root-hardenedvault opened the pull request:
The returned value of xmlnode_get_data() should be released with g_free(),
as stated in xmlnode.h of libpurple.

Signed-off-by: HardenedVault <root@hardenedvault.net>

EionRobb commented on src/lurch.c:
```diff
@@ -1599,8 +1600,8 @@ static void lurch_message_encrypt_groupchat(PurpleConnection * gc_p, xmlnode **
-      purple_conv_chat_write(chat_p, curr_muc_member_p->handle, xmlnode_get_data(body_node_p), PURPLE_MESSAGE_SEND, time((void *) 0));
+      body_data = xmlnode_get_data(body_node_p);
+      purple_conv_chat_write(chat_p, curr_muc_member_p->handle, body_data, PURPLE_MESSAGE_SEND, time((void *) 0));
       continue;
```
Doesn't the `g_free()` need to be before the continue?

root-hardenedvault replied:
Please review again. Thanks.

root-hardenedvault committed:
Fix misuse of xmlnode_get_data()

The returned value of xmlnode_get_data() should be released with g_free(),
as stated in xmlnode.h of libpurple.

Signed-off-by: HardenedVault <root@hardenedvault.net>

gkdr commented:
Thanks, good catch. the second instance is code I should remove soon, as it should be handled by the carbons plugin...
